yo/main.py

```diff
@@ -1508,6 +1508,37 @@ def run_for_instance(self, inst: YoInstance) -> None:
             )
 
 
+class CopyIdCmd(SingleInstanceCommand):
+    name = "copy-id"
+    description = "copy an SSH public key onto an instance using ssh-copy-id"
+
+    def add_args(self, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument("instance_name", type=str, help="Name of the instance")
+        parser.add_argument("-i", "--identity-file", type=str, required=True,
+                            help="Specify path to the public key file")
+
+    def run_for_instance(self, instance: YoInstance) -> None:
+
+        # Firstly, we need to extract instance name and public key file path from command-line arguments
+        instance_name = instance.name
+        public_key_file_path = self.args.identity_file
+
+        # Now, construct the command
+        ssh_copy_id_cmd = [
+            "ssh-copy-id",
+            "-i",
+            public_key_file_path,
+            f"{instance_name}@{instance.id}" # Assuming IP address is stored in 'id'
```

Comment:
The instance IP address is not stored in the `.id` field, that's the OCI instance ID. To get the instance IP address, you can use `self.c.get_instance_ip(instance)`.

Further, `instance_name` is the hostname of the instance. The SSH command expects that the part before the `@` sign is a username. You'll need to get the correct username to use based on the image (most images have a default username `opc`, but some have the default as `ubuntu`).

```suggestion
        instance_name = instance.name
        public_key_file_path = self.args.identity_file
        ip = self.c.get_instance_ip(instance)
        img = ctx.get_image(inst.image_id)
        user = OS_TO_USER[img.os]

        ssh_copy_id_cmd = [
            "ssh-copy-id",
            "-i",
            public_key_file_path,
            f"{user}@{ip}"
```

Reply:
checked and done